replicatedhq/ship
# Open a web browser when ship daemon starts

What I Did
------------

Ship now opens up a web browser

How I Did it
------------

- Add call to golang `open` lib for cross-platform browser opening
- Add `--no-open` flag to disable

Also, a bunch of tiny unrelated stuff stuff

- fix a deadlock in finalizeAppConfig by removing a call to `locker()()` in ShipDaemon
- add debug logging to `locker()()` in ShipDaemon
- ~refactor errChan off of `ShipDaemon` fields, and just create it in the only method that needs it~
- remove some random e2e testing stuff that is unused
- comment out a line that was breaking onprem ui
- add webpack write file plugin for `make build`ing with the output of WDS
- swap port 8800 with 8880 (WDS `make serve_ship` now listens on 8880)
- Add param for json format logging, default is still `key=value` format
- remove SHAs from goreleaser tag templates

Basically, https://www.youtube.com/watch?v=AbSehcT19u0

How to verify it
------------

build ship, then run `bin/ship init ...` or the ol' `make run` task, see browser open (wont work in container)

Description for the Changelog
------------

Update ship to open a browser when not running in a container

Picture of a Boat (not required but encouraged)
------------

:speedboat:
